crates/napi/src/next_api/project.rs

```diff
@@ -1284,24 +1282,29 @@ pub async fn get_source_map_rope(
         // Currently, this is inaccurate as it does not cover RSC server
         // chunks.
         map = container.get_source_map(client_path, module);
+        if map.await?.is_none() {
+            bail!("chunk/module '{}' is missing a sourcemap", file_path);
+        }
     }
 
-    if map.await?.is_none() {
-        bail!("chunk/module '{}' is missing a sourcemap", file_path);
-    }
+    Ok(map)
+}
 
-    Ok(Some(map))
+#[turbo_tasks::function(operation)]
+pub fn get_source_map_rope_operation(
+    container: ResolvedVc<ProjectContainer>,
+    file_path: RcStr,
+) -> Vc<OptionStringifiedSourceMap> {
+    get_source_map_rope(*container, file_path)
 }
 
-pub async fn get_source_map(
-    container: Vc<ProjectContainer>,
-    file_path: String,
-) -> Result<Option<ReadRef<OptionSourceMap>>> {
-    let Some(map) = get_source_map_rope(container, file_path).await? else {
-        return Ok(None);
-    };
```

Comment:
`SourceMap::new_from_rope_cached` duplicates this check for `None`, so I just removed it.